zendesk_tickets/views.py

```diff
@@ -14,11 +23,31 @@ def ticket(request,
     if request.method == 'POST':
         form = form_class(request.POST)
         if form.is_valid():
-            form.submit_ticket(request, subject, tags, ticket_template_name)
-            return HttpResponseRedirect(success_redirect_url)
+            try:
+                form.submit_ticket(request, subject, tags, ticket_template_name)
+                referer = form.cleaned_data.get('referer')
+                if referer:
+                    come_from = urlparse(referer).path
+                    success_redirect_url = (
+                        success_redirect_url +
+                        ('?%s=%s' % (RETURN_URL_FIELD, come_from))
+                    )
+                return HttpResponseRedirect(success_redirect_url)
+            except HTTPError:
+                form.add_error(NON_FIELD_ERRORS, _('Unexpected error.'))
     else:
         form = form_class(
             initial={'referer': request.META.get('HTTP_REFERER')}
         )
 
     return render(request, template_name, {'form': form})
+
+
+def success(request, template_name=None):
+    return_to = request.GET.get(RETURN_URL_FIELD)
+
+    # Ensure the user-originating redirection url is safe.
+    if not return_to or not is_safe_url(url=return_to, host=request.get_host()):
+        return_to = resolve_url('/')
```

Comment:
if these shared packages are to be very generic, i think perhaps `None` might be a better choice (and hence the template needing an `{% if return_to %}` wrapper around some text) because `'/'` may not be an appropriate url in the general case?